fili-core/src/main/java/com/yahoo/bard/webservice/web/filters/FiliTimingFilter.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.filters;
+
+import com.yahoo.bard.webservice.logging.RequestLogUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import javax.annotation.Priority;
+import javax.inject.Singleton;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.MultivaluedMap;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+/**
+ * Starts/stops the timing for a request as well as stashing the request-id header in the RequestLog. This filter
+ * is useful when you <b>do not</b> want to use the {@link BardLoggingFilter}.
+ */
+@PreMatching
+@Singleton
+@Priority(1)
+public class FiliTimingFilter implements ContainerResponseFilter, ContainerRequestFilter {
```

Comment:
This could really use a better name, since it isn't just doing timing (it's also stashing the request-id header). Maybe `FiliInitializationFilter`?